### PR TITLE
make device transfers non-blocking in transforms

### LINKED ISSUE #7504: Normalize and RandomMixup contain tensor creation without non_blocking=True
### 🐛 Describe the bug

In `torchvision.transforms.Normalize`, the mean and std move to gpu is blocking:

https://github.com/pytorch/vision/blob/289fce29b3e2392114aadbe7a419df0f2e3ac1be/torchvision/transforms/functional_tensor.py#L947-L948

In `RandomMixup`, the dirichlet argument move to gpu is blocking:

https://github.com/pytorch/vision/blob/289fce29b3e2392114aadbe7a419df0f2e3ac1be/references/classification/transforms.py#L71

### Versions

```
PyTorch version: 2.0.0+cu117                                                                                                                                                                       [85/1856]
Is debug build: False                                                                                                                                                                                       
CUDA used to build PyTorch: 11.7                                                                                                                                                                            
ROCM used to build PyTorch: N/A                                                                                                                                                                             
                                                                                                                                                                                                            
OS: Red Hat Enterprise Linux release 8.6 (Ootpa) (x86_64)                                                                                                                                                   
GCC version: (GCC) 8.5.0 20210514 (Red Hat 8.5.0-10)                                                                                                                                                        
Clang version: Could not collect                                                                                                                                                                            
CMake version: version 3.26.1                                                                                                                                                                               
Libc version: glibc-2.28                                                                                                                                                                                    
                                                                                                                                                                                                            
Python version: 3.11.0 (main, Mar  1 2023, 18:26:19) [GCC 11.2.0] (64-bit runtime)                                                                                                                          
Python platform: Linux-4.18.0-372.41.1.el8_6.x86_64-x86_64-with-glibc2.28                                                                                                                                   
Is CUDA available: False                                                                                                                                                                                    
CUDA runtime version: No CUDA                                                                                                                                                                               
CUDA_MODULE_LOADING set to: N/A                                                                                                                                                                             
GPU models and configuration: No CUDA                                                                                                                                                                       
Nvidia driver version: No CUDA                                                                                                                                                                              
cuDNN version: No CUDA                                                                                                                                                                                      
HIP runtime version: N/A                                                                                                                                                                                    
MIOpen runtime version: N/A                                                                                                                                                                                 
Is XNNPACK available: True                                                                                                                                                                                  
                                                                                                                                                                                                            
CPU:                                                                                                                                                                                                        
Architecture:        x86_64                                                                                                                                                                                 
CPU op-mode(s):      32-bit, 64-bit                                                                                                                                                                         
Byte Order:          Little Endian                                                                                                                                                                          
CPU(s):              80                                                                                                                                                                                     
On-line CPU(s) list: 0-79                                                                                                                                                                                   
Thread(s) per core:  2                                                                                                                                                                                      
Core(s) per socket:  20                                                                                                                                                                                     
Socket(s):           2                                                                                                                                                                                      
NUMA node(s):        2                                                                                                                                                                                      
Vendor ID:           GenuineIntel                                                                                                                                                                           
CPU family:          6                                                                                                                                                                                      
Model:               85                                                                                                                                                                                     
Model name:          Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz                                                                                                                                               
Stepping:            7
CPU MHz:             2500.000
BogoMIPS:            5000.00
L1d cache:           32K
L1i cache:           32K
L2 cache:            1024K
L3 cache:            28160K
NUMA node0 CPU(s):   0-19,40-59
NUMA node1 CPU(s):   20-39,60-79
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs
 bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xs
ave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm mpx rd
t_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts pku ospke
avx512_vnni md_clear flush_l1d arch_capabilities

Versions of relevant libraries:
[pip3] eztorch==0.3.0
[pip3] numpy==1.24.2
[pip3] pytorch-lightning==1.9.4
[pip3] pytorchvideo==0.1.5
[pip3] torch==2.0.0
[pip3] torchmetrics==0.11.4
[pip3] torchvision==0.15.1
[pip3] triton==2.0.0
[conda] eztorch                   0.3.0                     dev_0    <develop>
[conda] numpy                     1.24.2                   pypi_0    pypi
[conda] pytorch-lightning         1.9.4                    pypi_0    pypi
[conda] pytorchvideo              0.1.5                    pypi_0    pypi
[conda] torch                     2.0.0                    pypi_0    pypi
[conda] torchmetrics              0.11.4                   pypi_0    pypi
[conda] torchvision               0.15.1                   pypi_0    pypi
[conda] triton                    2.0.0                    pypi_0    pypi
```

### PR DESCRIPTION
**Pull request by nlgranger**
Fixes #7504

### CODE REVIEW

**Comment by NicolasHug** [comment]
Thanks for the PR @nlgranger but I am not certain that the use of `non_blocking=True` will have any significant effect in a lot of these cases: to be non-blocking, the data transfer must come from **pinned** memory, and it looks like a lot of those tensors are in pageable memory.

**Comment by nlgranger** [comment]
I forgot about pinning memory... 
In my own code I have opted for the second route, which consists in registering parameters as nn.Module buffers. It breaks the API but would make a lot more sense now that that Transfoms are Modules as well and that many of us are running transforms on GPU.

**Comment by nlgranger** [comment]
I have fixed a few remaining bugs. In the end the API is unchanged. As mentioned previously, I have no idea how to implement tests for this. I don't think it makes sense to check for sync point globally since some ops might be unavoidable (ex: nonzero, unique).
